pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -138,6 +138,28 @@ func (r *replicaRaftStorage) InitialState() (raftpb.HardState, raftpb.ConfState,
 	return hs, cs, nil
 }
 
+// TODO(pav-kv): remove the adapter methods.
```

Comment:
One way to do it: don't have a single `raft.Storage` interface. Use `raft.LogStorage` and `raft.StateStorage` separately.